### ISSUE TITLE
$options of loadModalAsync function

### ISSUE THREAD

**Issue by imnicy** [orchidsoftware/platform#958]
loadModalAsync($modal, $method, $options, $text)

when the $options  is a string attribute in model, the modal cannot work.

file: orchid/platform/resources/views/partials/td/async.blade.php
line: 6
original:   data-modal-params="@json($attributes)"
after change:  data-modal-params=@json($attributes)

through the above steps, my problem was solved.

**Comment by tabuna** [comment]
Hi. You have closed the question. 

Has it become irrelevant? Did you solve your problem differently?

**Comment by imnicy** [comment]
I found two minor problems when i use the modal layout, 
suggestion 1:     data-modal-params="@json($attributes)"    should be change to    data-modal-params=@json($attributes)   at file: resources/views/partials/td/async.blade.php
suggestion 2:      when i use the modal, Through the following code:
     public function layout(): array    {        return [            OptionEditLayout::class,            OptionTableLayout::class,
            Layout::modal('oneAsyncModal', [
                OptionModalEditLayout::class,
            ])->async('asyncGetOption'),        ];    }
    When I click the modal wake-up button, the layout slug always matched "OptionEditLayout::class", so, this code doesn't work properly.    Then I changed the code on [src/Screen/Layouts/Base.php], at line 169-172, to:
    public function getSlug(): string    {        return sha1(serialize(new \ReflectionObject($this)));    }
    It works, and it can matched the OptionModalEditLayout::class.  ^ - ^
----- 原始邮件 -----
发件人：Alexandr Chernyaev <notifications@github.com>
收件人：orchidsoftware/platform <platform@noreply.github.com>
抄送人：imnicy <imnicy@sina.com>,  State change <state_change@noreply.github.com>
主题：Re: [orchidsoftware/platform] $options of loadModalAsync function (#958)
日期：2019年09月19日 15点15分

Hi. You have closed the question.
Has it become irrelevant? Did you solve your problem differently?

**Comment by imnicy** [comment]
I found two minor problems when i use the modal layout, 
suggestion 1:    
 ```data-modal-params="@json($attributes)" ```   
 should be change to    
 ```data-modal-params=@json($attributes)```
 at file: resources/views/partials/td/async.blade.php

suggestion 2:      
when i use the modal, Through the following code:
 ```php
public function layout(): array    
{        
    return [           
        OptionEditLayout::class,           
        OptionTableLayout::class,
        Layout::modal('oneAsyncModal', [
                OptionModalEditLayout::class,
        ])->async('asyncGetOption'),       
     ];    
}
```
when I click the modal wake-up button, the layout slug always matched "OptionEditLayout::class", so, this code doesn't work properly.    Then I changed the code on [src/Screen/Layouts/Base.php], at line 169-172, to:
```php
public function getSlug(): string    {        
    return sha1(serialize(new \ReflectionObject($this)));    
}
```
It works, and it can matched the OptionModalEditLayout::class.  ^ - ^

**Comment by tabuna** [comment]
I propose then to change:
```
data-modal-params="@json($attributes)"
```
to
```
data-modal-params='@json($attributes)'
```

This should fix and prevent unpleasant situations.

Regarding the second error, can you give me more information? Are you using the latest version?

**Comment by imnicy** [comment]
The second problem has been fixed and I tested it with version 6.1. Perfect.
